jwtauth.go

```diff
@@ -147,11 +148,22 @@ func (ja *JWTAuth) Decode(tokenString string) (t *jwt.Token, err error) {
 	return
 }
 
+// if the interface is itself a function call that function with the token
+// and return its results. this allows for use cases like jwks without
+// changing the api
+func keyFuncOrValue(i interface{}, t *jwt.Token) (interface{}, error) {
```

Comment:
IMHO, this case is not about reflection usage, I think, we can pass function explicitly, and make type for this function